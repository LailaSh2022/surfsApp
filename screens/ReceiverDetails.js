import React from 'react';
import { StyleSheet, View, Text, Pressable } from 'react-native';
import {
    StyledBackButton,
    ButtonText,
    StyledContainer,
    StyledButton,
} from "./../components/Styles";


const ReceiverDetails = () => {

    
    
    return (
        <StyledContainer>
            <View style={{ flexDirection: 'row' }}>
                <StyledBackButton>
                    <ButtonText>{'<'}</ButtonText>
                </StyledBackButton>
                <Text style={styles.headline}> Receiver Details</Text>
            </View>
            <View style={{ flexDirection: 'row', marginTop: 30 }}>
                <Text style={{width:'25%'} }>
                    Name:
                </Text>
                <Text> tatiane</Text>
            </View>
            <View style={{ flexDirection: 'row', marginTop: 15 }}>
                <Text style={{ width: '25%' }} >
                    Account No: 
                </Text>
                <Text> 001-11111111111111-07</Text>
            </View>
            <View style={{ flexDirection: 'row', marginTop: 15 }}>
                <Text style={{ width: '25%' }}>
                    Address:
                </Text>
                <Text> 10 Martin Road Orakei Auckland {`\n`} New Zealand</Text>
            </View>
            <View style={{ flexDirection: 'row', marginTop: 15 }}>
                <Text style={{ width: '25%' }}>
                    Email:
                </Text>
                <Text> tatian@gmail.com</Text>
            </View>
            <View style={{ flexDirection: 'row', marginTop: 15 }}>
                <Text>
                    Bank Info:
                </Text>
            </View>
            <View style={{ flexDirection: 'row', marginTop: 15 }}>
                <Text style={{ width: '25%' }}>
                    Name: 
                </Text>
                <Text> XYZ Bank New Zealand LTD</Text>
            </View>
            <View style={{ flexDirection: 'row', marginTop: 15 }}>
                <Text style={{ width: '25%' }}>
                    Address:
                </Text>
                <Text> Wiri Station RD, Manukau, Auckland 2104, {'\n'} New Zealand</Text>
            </View>
            <View style={{ flexDirection: 'row', marginTop: 15 }}>
                <Text style={{ width: '25%' }}>
                    Swift Code:
                </Text>
                <Text style={{ paddingBottom:30 }}> XYZ BNZ 23</Text>
            </View>
            <StyledButton>
                <ButtonText >Continue</ButtonText>
            </StyledButton>
        </StyledContainer>
        

    );
};


const styles = StyleSheet.create({

    headline: {
        flex:2,
        textAlign: 'center',
        fontWeight: 'bold',
        fontSize: 18,
        paddingTop: 10,
        color:'#152A5F'
    }
});

export default ReceiverDetails;