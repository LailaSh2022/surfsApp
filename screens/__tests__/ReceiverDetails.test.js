import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import ReceiverDetails from "../ReceiverDetails";

it(
  "Testing ReceiverDetails screen against its snapshot. " +
    "If there is any unexpected change in the screen, it will prompt test failed.",
  () => {
    const mockRoute = {
      params: {
        receiver: {
          FirstName: "John",
          MiddleName: "",
          LastName: "Dear",
          Bank_Account_Number: "07-111111111111111-02",
          Address: "10 Martin Road, Manurewa, Auckland 2102",
          Email: "john_deer@gmail.com",
          Swift_Code: "012",
        },
      },
    };

    const tree = renderer
      .create(<ReceiverDetails route={mockRoute} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  }
);
