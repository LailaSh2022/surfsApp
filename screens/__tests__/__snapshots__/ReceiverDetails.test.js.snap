// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ReceiverDetails screen against its snapshot. If there is any unexpected change in the screen, it will prompt test failed. 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 25,
        "paddingLeft": 25,
        "paddingRight": 25,
        "paddingTop": 64,
      },
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#152A5F",
          "borderRadius": 5,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#ffffff",
              "fontSize": 16,
              "textAlign": "center",
            },
          ]
        }
      >
        &lt;
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#152A5F",
          "flex": 2,
          "fontSize": 18,
          "fontWeight": "bold",
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
       Receiver Details
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 30,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Name:
    </Text>
    <Text>
      John
       
       
      Dear
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Account No:
    </Text>
    <Text>
      07-111111111111111-02
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Address:
    </Text>
    <Text
      style={
        {
          "paddingRight": "15%",
        }
      }
    >
      10 Martin Road, Manurewa, Auckland 2102
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Email:
    </Text>
    <Text>
      john_deer@gmail.com
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "width": "25%",
        }
      }
    >
      Swift Code:
    </Text>
    <Text
      style={
        {
          "paddingBottom": 30,
        }
      }
    >
      012
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#152A5F",
        "borderRadius": 5,
        "height": 60,
        "justifyContent": "center",
        "marginVertical": 5,
        "opacity": 1,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
            "fontSize": 16,
            "textAlign": "center",
          },
        ]
      }
    >
      Continue
    </Text>
  </View>
</View>
`;
